packages/shared/src/components/modals/common/Modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+import ReactModal from 'react-modal';
+import classNames from 'classnames';
+import { ModalHeader } from './ModalHeader';
+import { ModalBody } from './ModalBody';
+import { ModalFooter } from './ModalFooter';
+import { ModalSidebar } from './ModalSidebar';
+import { ModalKind, ModalPropsContext, ModalSize } from './types';
+
+export type ModalProps = ReactModal.Props & {
+  children: React.ReactNode;
+  kind?: ModalKind;
+  size?: ModalSize;
+};
+
+export function Modal({
+  className,
+  overlayClassName,
+  children,
+  kind = ModalKind.FlexibleCenter,
+  size = ModalSize.Medium,
+  onRequestClose,
+}: ModalProps): ReactElement {
+  const modalKindToOverlayClassName = {
+    [ModalKind.FixedCenter]: 'justify-center',
+    [ModalKind.FlexibleCenter]: 'justify-center',
+    [ModalKind.FlexibleTop]: '',
+  };
+  const modalKindToClassName = {
+    [ModalKind.FixedCenter]: '',
+    [ModalKind.FlexibleCenter]: 'h-[40rem]',
+    [ModalKind.FlexibleTop]: classNames(
+      'mt-10',
+      size === ModalSize.Medium && 'mt-20',
+      size === ModalSize.Large && 'mt-14',
+    ),
+  };
+  const modalSizeToClassName = {
+    [ModalSize.XSmall]: 'max-w-[21.25]',
+    [ModalSize.Small]: 'max-w-[26.25rem]',
+    [ModalSize.Medium]: 'max-w-[35rem]',
+    [ModalSize.Large]: 'max-w-[63.75rem]',
+  };
+  return (
+    <ReactModal
+      isOpen
+      overlayClassName={classNames(
+        'overlay flex fixed flex-col inset-0 max-h-[100vh] items-center px-5 bg-gradient-to-r to-theme-overlay-to from-theme-overlay-from z-[10]',
+        overlayClassName,
+        modalKindToOverlayClassName[kind],
+      )}
+      className={classNames(
+        'focus:outline-none modal flex flex-col relative w-full max-h-[calc(100vh-5rem)] overflow-y-auto items-center bg-theme-bg-tertiary shadow-2 border border-theme-divider-secondary rounded-16',
+        className,
+        modalKindToClassName[kind],
+        modalSizeToClassName[size],
+      )}
+    >
+      <ModalPropsContext.Provider value={{ size, kind, onRequestClose }}>
+        {children}
+      </ModalPropsContext.Provider>
+    </ReactModal>
+  );
+}
+
+Modal.Size = ModalSize;
+Modal.Kind = ModalKind;
+Modal.Header = ModalHeader;
+Modal.Body = ModalBody;
+Modal.Footer = ModalFooter;
+Modal.Sidebar = ModalSidebar;
```

Comment:
Why do we choose the dot notation?
Not against it, just something we haven't used anywhere else in favor of things like ModalSize.

Reply:
So that we can import and use everything modal related with the same class.

Reply:
Would you not favor changing to `export ModalSize`?
You can still import everything from the same file.
Another common practice is to have an index.ts file in the directory that only exports whatever is necessary.
Something like this: https://github.com/dailydotdev/pulumi-common/blob/main/src/index.ts
Both ways are somewhat opinionated so whatever works for you works for me :)

Reply:
Yeah, that was my first idea as well, but I went with this approach because it is much easier to use the component.
You just import the Modal and all the child components will be in your fingertips. We talked with @rebelchris about this and decided to give it a try, happy to jump on a call if you want more context or have any feedback about it.

Reply:
This was just my two cents. You make the call 🤙
Thanks for the info